File: LeetCode/Easy/0704_Binary_Search/Solution.go
package main

// 2022-03-25
func search(nums []int, target int) int {
	lo, hi := 0, len(nums)-1

	for lo <= hi {
		mid := lo + (hi-lo)/2

		if nums[mid] < target {
			lo = mid + 1
		} else if nums[mid] > target {
			hi = mid - 1
		} else {
			return mid
		}
	}

	return -1
}

File: LeetCode/Easy/0704_Binary_Search/Solution.cpp
// 2023-03-31
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int left = 0, right = nums.size()-1;
        while (left < right) {
            int mid = left + (right-left)/2;
            if (nums[mid] < target) {
                left = mid+1;
            } else {
                right = mid;
            }
        }
        return (nums[left] == target ? left : -1);
    }
};